grafana-plugin/src/models/outgoing_webhook/outgoing_webhook.ts

```diff
@@ -43,18 +46,34 @@ export class OutgoingWebhookStore extends BaseStore {
 
   @action
   async updateItem(id: OutgoingWebhook['id'], fromOrganization = false) {
-    const response = await this.getById(id, false, fromOrganization);
+    let outgoungWebhook;
+
+    try {
+      outgoungWebhook = await this.getById(id, true, fromOrganization);
+    } catch (error) {
+      if (error.response.data.error_code === 'wrong_team') {
+        outgoungWebhook = {
+          id,
+          name: '🔒 Private outgoing webhook',
+          private: true,
+        };
```

Comment:
can the logic to create this object be be extracted/reused? We're doing something similar [here](https://github.com/grafana/oncall/pull/1528/files#diff-ec407c4a628388542c8d8826e89b579e7f9d99ff085ac263050d3e547ba8c8bfR80-R84) and [here](https://github.com/grafana/oncall/pull/1528/files#diff-5ad2c4830bdc85000eb867815c71d2b6aa5dbcbd4aedc28ad65d9d7f4dd09393R161-R164) (and maybe in a few other spots in the PR). Something like this:

```ts
type MyObject = {
  id: string;  // or maybe number? not sure
  objectName: string;
};

interface PrivateObject extends Pick<MyObject, 'id'> {
  name: string;
  private: boolean;
}

const generatePrivateObject = ({ id, objectName }: MyObject): PrivateObject => ({
  id,
  name `🔒 Private ${objectName}`,
  private: true,
});
```